pallets/alt-vr/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+
+            // Create Altvr data
+            Self::new_altvr(&owner, &username, &vchatname);
+
+            let event = Event::AltvrCreated(owner, username, vchatname);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+
+        /// Update Altvr username of the caller
+        #[pallet::weight(T::WeightInfo::update_username())]
+        #[transactional]
+        pub fn update_username(
+            origin: OriginFor<T>,
+            username: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = username.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = username.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongUsername);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.username = username.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrUsernameUpdated(owner, username);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+
+        /// Update Altvr vchatname of the caller
+        #[pallet::weight(T::WeightInfo::update_vchatname())]
+        #[transactional]
+        pub fn update_vchatname(
+            origin: OriginFor<T>,
+            vchatname: ternoa::String,
+        ) -> DispatchResultWithPostInfo {
+            let owner = ensure_signed(origin)?;
+
+            let username_lower_bound = vchatname.len() >= T::MinNameLength::get() as usize;
+            let username_upper_bound = vchatname.len() <= T::MaxNameLength::get() as usize;
+
+            ensure!(username_lower_bound, Error::<T>::TooShortUsername);
+            ensure!(username_upper_bound, Error::<T>::TooLongVchatname);
+
+            Altvrs::<T>::try_mutate(owner.clone(), |res| -> Result<(), Error<T>> {
+                let altvrdata = res.as_mut().ok_or(Error::<T>::UserNotFound)?;
+                altvrdata.vchatname = vchatname.clone();
+                Ok(())
+            })?;
+
+            let event = Event::AltvrVchatnameUpdated(owner, vchatname);
+            Self::deposit_event(event);
+
+            Ok(().into())
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// Altvr was created. \[owner, username, vchatname\]
+        AltvrCreated(T::AccountId, ternoa::String, ternoa::String),
+        /// Altvr username updated  \[owner, username\]
+        AltvrUsernameUpdated(T::AccountId, ternoa::String),
+        /// Altvr vchatname updated  \[owner, vchatname\]
+        AltvrVchatnameUpdated(T::AccountId, ternoa::String),
```

Comment:
Instead of `AltvrVchatnameUpdated` use `AltvrVchatnameChanged`.

Reply:
done